pyccel/parser/semantic.py

```diff
@@ -708,6 +706,11 @@ def _handle_function_args(self, arguments, **settings):
             if isinstance(a.value, StarredArguments):
                 args.extend([FunctionCallArgument(av) for av in a.value.args_var])
             else:
+                if isinstance(a.value, PyccelArithmeticOperator) and a.value.rank:
+                    tmp_var = PyccelSymbol(self.scope.get_new_name(), is_temp=True)
+                    assign = self._visit(Assign(tmp_var, arg.value, fst= arg.value.fst))
+                    self._additional_exprs[-1].append(assign)
+                    a = FunctionCallArgument(self._visit(tmp_var))
```

Comment:
Please add a test for this case

Reply:
I added some tests in [2158507](https://github.com/pyccel/pyccel/pull/1187/commits/215850701c290467e536c5c1bae4755cc9b1c38a), is it alright to add the tests in scripts instead of epyccel, because i wanted to add tests for arithmetic operations with functions return.

Reply:
Tests in the `tests/pyccel/` folder are fine, but you need to add them to the `test_pyccel.py` file in order for them to be executed. I don't see any tests testing this though. Is it not supposed to handle things like:
```
f(a+b)
```
?

Reply:
Oh i guess i didnt add em, i was testing em in already defined functions like in `np.array` is it fine ?

Reply:
@EmilyBourne i did add a test with an arithmetic add in [fdf0609](https://github.com/pyccel/pyccel/pull/1187/commits/fdf06091e95635660e96f393a40e239bf52e2805).

Reply:
> Oh i guess i didnt add em, i was testing em in already defined functions like in np.array is it fine ?

I don't see this test. Can you point me to the line

Reply:
Never mind I found it. For some reason I thought this code only applied to user defined functions